crates/network-scanner/src/task_utils.rs

```diff
@@ -0,0 +1,174 @@
+use std::sync::atomic::{AtomicBool, AtomicUsize};
+use std::{net::IpAddr, sync::Arc, time::Duration};
+
+use dashmap::DashMap;
+
+use futures::Future;
```

Comment:
issue: The `Future` trait is now part of `std`
```suggestion
use std::future::Future;
```